### PR TITLE
ScKeynodes.rrel(index: int)

### PR DESCRIPTION
**Pull request by FalseR20**
Role nodes `rrel_1`, `rrel_2`, `rrel_i` are used very often.
So, I added method to ScKeynodes for easy usage:

```python
ScKeynodes.rrel(index: int) -> ScAddr
```

It guarantees valid rrel node.

UPD: max rrel index equal 10 and can be changed

### CODE REVIEW

File: tests/test_keynodes.py [hunk 1]
```diff
@@ -38,8 +38,11 @@ def test_keynodes_initialization(self):
...
-        self.assertTrue(rrel_21.is_valid())
-        self.assertTrue(check_elements(rrel_21)[0].is_role())
+        rrel_1 = ScKeynodes.rrel(1)
+        self.assertTrue(rrel_1.is_valid())
+        self.assertTrue(check_elements(rrel_1)[0].is_role())
+
+    def test_large_rrel(self):
+        self.assertRaises(KeyError, ScKeynodes.rrel, ScKeynodes.max_rrel_index + 1)
+        ScKeynodes.max_rrel_index += 1
+        self.assertTrue(ScKeynodes.rrel(11).is_valid())
```
Comment (AlexandrZagorskiy) [inline tests/test_keynodes.py:48]: Dont give a possibility to change max_rrel_index. I think this should be const value. So 47 and 48 lines should be removed

**Review by AlexandrZagorskiy — COMMENTED** [review]
Update changelog

File: tests/test_keynodes.py (hunk 1 above)
Comment (FalseR20) [reply tests/test_keynodes.py:48]: Why do not give possibility to change it?

Comment (AlexandrZagorskiy) [reply tests/test_keynodes.py:48]: It's bad idea to create a lots of different nodes with indexes, better to use ordered set structure

Comment (FalseR20) [reply tests/test_keynodes.py]: Made it private

File: src/sc_kpm/sc_keynodes.py [hunk 2]
```diff
@@ -56,6 +57,12 @@ def resolve(cls, identifier: Idtf, sc_type: Optional[ScType]) -> ScAddr:
             cls._logger.debug("Resolved %s identifier with type %s: %s", repr(identifier), repr(sc_type), repr(addr))
         return addr
 
+    def rrel(cls, index: int) -> ScAddr:
+        """Get rrel_i node. Max rrel index is 10"""
+        if index > cls._max_rrel_index:
+            raise KeyError(f"You cannot use rrel more than {cls._max_rrel_index}")
+        return cls.resolve(f"rrel_{index}", NODE_CONST_ROLE)  # pylint: disable=no-value-for-parameter
+
```
Comment (AlexandrZagorskiy) [inline src/sc_kpm/sc_keynodes.py:65]: role relation can be used not only with indexes. So if you want to use it only for indexes -- rename this, maybe to the _rrel_index()_ or smth like that. Also add check for input value, right now I can pass a string here

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: tests/test_keynodes.py [hunk 3]
```diff
@@ -36,3 +36,11 @@ def test_delete_keynode(self):
...
     def test_keynodes_initialization(self):
         self.assertRaises(TypeError, ScKeynodes)
+
+    def test_rrel(self):
+        rrel_1 = ScKeynodes.rrel(1)
+        self.assertTrue(rrel_1.is_valid())
+        self.assertTrue(check_elements(rrel_1)[0].is_role())
+
+    def test_large_rrel(self):
+        self.assertRaises(KeyError, ScKeynodes.rrel, ScKeynodes._max_rrel_index + 1)
```
Comment (AlexandrZagorskiy) [inline tests/test_keynodes.py:46]: add test for wrong input value

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: src/sc_kpm/sc_keynodes.py (hunk 2 above)
Comment (FalseR20) [reply src/sc_kpm/sc_keynodes.py]: Done

File: tests/test_keynodes.py (hunk 3 above)
Comment (FalseR20) [reply tests/test_keynodes.py]: Done

**Comment by FalseR20** [comment]
@AlexandrZagorskiy @NikitaZotov 
Review please

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**